net/ipv4/fib_semantics.c

```diff
@@ -258,9 +259,7 @@ static inline int nh_comp(const struct fib_info *fi, const struct fib_info *ofi)
 		if (nh->nh_oif != onh->nh_oif ||
 		    nh->nh_gw  != onh->nh_gw ||
 		    nh->nh_scope != onh->nh_scope ||
-#ifdef CONFIG_IP_ROUTE_MULTIPATH
```

Comment:
The field: nh_weight is conditionally added to struct fip_nh by #ifdef CONFIG_IP_ROUTE_MULTIPATH above.

Do you need to keep the #ifdef here as well?